src/test/ui/liveness/liveness-upvars.rs

```diff
@@ -0,0 +1,85 @@
+// check-pass
+#![crate_type = "lib"]
+#![warn(unused)]
+
+pub fn f0() {
+    let mut last = None;
+    let mut f = move |s| {
+        last = Some(s); //~  WARN value assigned to `last` is never read
+                        //~| WARN variable `last` is assigned to, but never used
```

Comment:
isn't this... false? Oh, I guess it's true because of the `move`

Reply:
I wonder if we can provide some 'help" text or something about that

Reply:
can we add a test like this, except that the closure is *not* `move`?

Reply:
As for an additional explanation, maybe something along the lines of: "variable captured by value, did you mean to capture by reference instead?", possibly highlighting the "move" part of closure.